stdlib/public/SwiftShims/swift/shims/CMakeLists.txt

```diff
@@ -94,7 +91,8 @@ if(NOT SWIFT_INCLUDE_TOOLS AND
   endif()
   message(STATUS "Using clang Resource Directory: ${clang_headers_location}")
 else()
-  set(clang_headers_location "${LLVM_LIBRARY_OUTPUT_INTDIR}/clang/${CLANG_VERSION}")
+  # ... or the one we just built
+  set(clang_headers_location "${LLVM_LIBRARY_OUTPUT_INTDIR}/clang/${LLVM_VERSION_MAJOR}")
```

Comment:
Good point -- I wrongly assumed they are set to the same value, but that could not be the case (as per https://github.com/llvm/llvm-project/commit/ebfc680685b7bb246d636cc9305ae6ab268eb0b5)

Since `CLANG_VERSION_MAJOR` is not reexported by `ClangConfig.cmake`, I need to update the check before as well to ensure we specify that when configuring the Swift build system.

Addressed this in 6b3ecfbc939704793de0532d7d9eb5c7cf0edba6